src/machine/asm/execute_aarch64.S

```diff
@@ -0,0 +1,1633 @@
+  ldr RS2, REGISTER_ADDRESS(RS2)
+  sub RS1, RS1, RS2
+  WRITE_RD(RS1)
+  NEXT_INST
+.CKB_VM_ASM_LABEL_OP_SUBW:
+  DECODE_R
+  ldr RS1, REGISTER_ADDRESS(RS1)
+  ldr RS2, REGISTER_ADDRESS(RS2)
+  sub RS1, RS1, RS2
+  sxtw RS1, RS1w
+  WRITE_RD(RS1)
+  NEXT_INST
+.CKB_VM_ASM_LABEL_OP_SW:
+  DECODE_S
+  ldr RS1, REGISTER_ADDRESS(RS1)
+  add RS1, RS1, IMMEDIATE
+  CHECK_WRITE(RS1, 4)
+  add RS1, RS1, CKB_VM_ASM_ASM_CORE_MACHINE_OFFSET_MEMORY
+  ldr RS2, REGISTER_ADDRESS(RS2)
+  str RS2w, [MACHINE, RS1]
+  NEXT_INST
+.CKB_VM_ASM_LABEL_OP_XOR:
+  DECODE_R
+  ldr RS1, REGISTER_ADDRESS(RS1)
+  ldr RS2, REGISTER_ADDRESS(RS2)
+  eor RS1, RS1, RS2
+  WRITE_RD(RS1)
+  NEXT_INST
+.CKB_VM_ASM_LABEL_OP_XORI:
+  DECODE_I
+  ldr RS1, REGISTER_ADDRESS(RS1)
+  eor RS1, RS1, IMMEDIATE
+  WRITE_RD(RS1)
+  NEXT_INST
+.CKB_VM_ASM_LABEL_OP_ADDUW:
+  DECODE_R
+  ldr RS1, REGISTER_ADDRESS(RS1)
+  ldr RS2, REGISTER_ADDRESS(RS2)
+  mov RS1w, RS1w
+  add RS1, RS1, RS2
+  WRITE_RD(RS1)
+  NEXT_INST
+.CKB_VM_ASM_LABEL_OP_ANDN:
+  DECODE_R
+  ldr RS1, REGISTER_ADDRESS(RS1)
+  ldr RS2, REGISTER_ADDRESS(RS2)
+  mvn RS2, RS2
+  and RS1, RS1, RS2
+  WRITE_RD(RS1)
+  NEXT_INST
+.CKB_VM_ASM_LABEL_OP_BCLR:
+  DECODE_R
+  ldr RS1, REGISTER_ADDRESS(RS1)
+  ldr RS2, REGISTER_ADDRESS(RS2)
+  mov TEMP1, 1
+  lsl TEMP1, TEMP1, RS2
+  mvn TEMP1, TEMP1
+  and RS1, RS1, TEMP1
+  WRITE_RD(RS1)
+  NEXT_INST
+.CKB_VM_ASM_LABEL_OP_BCLRI:
+  DECODE_I
+  ldr RS1, REGISTER_ADDRESS(RS1)
+  mov TEMP1, 1
+  lsl TEMP1, TEMP1, IMMEDIATE
+  mvn TEMP1, TEMP1
+  and RS1, RS1, TEMP1
+  WRITE_RD(RS1)
+  NEXT_INST
+.CKB_VM_ASM_LABEL_OP_BEXT:
+  DECODE_R
+  ldr RS1, REGISTER_ADDRESS(RS1)
+  ldr RS2, REGISTER_ADDRESS(RS2)
+  lsr RS1, RS1, RS2
+  and RS1, RS1, 1
+  WRITE_RD(RS1)
+  NEXT_INST
+.CKB_VM_ASM_LABEL_OP_BEXTI:
+  DECODE_I
+  ldr RS1, REGISTER_ADDRESS(RS1)
+  lsr RS1, RS1, IMMEDIATE
+  and RS1, RS1, 1
+  WRITE_RD(RS1)
+  NEXT_INST
+.CKB_VM_ASM_LABEL_OP_BINV:
+  DECODE_R
+  ldr RS1, REGISTER_ADDRESS(RS1)
+  ldr RS2, REGISTER_ADDRESS(RS2)
+  mov TEMP1, 1
+  lsl TEMP1, TEMP1, RS2
+  eor RS1, RS1, TEMP1
+  WRITE_RD(RS1)
+  NEXT_INST
+.CKB_VM_ASM_LABEL_OP_BINVI:
+  DECODE_I
+  ldr RS1, REGISTER_ADDRESS(RS1)
+  mov TEMP1, 1
+  lsl TEMP1, TEMP1, IMMEDIATE
+  eor RS1, RS1, TEMP1
+  WRITE_RD(RS1)
+  NEXT_INST
+.CKB_VM_ASM_LABEL_OP_BSET:
+  DECODE_R
+  ldr RS1, REGISTER_ADDRESS(RS1)
+  ldr RS2, REGISTER_ADDRESS(RS2)
+  mov TEMP1, 1
+  lsl TEMP1, TEMP1, RS2
+  orr RS1, RS1, TEMP1
+  WRITE_RD(RS1)
+  NEXT_INST
+.CKB_VM_ASM_LABEL_OP_BSETI:
+  DECODE_I
+  ldr RS1, REGISTER_ADDRESS(RS1)
+  mov TEMP1, 1
+  lsl TEMP1, TEMP1, IMMEDIATE
+  orr RS1, RS1, TEMP1
+  WRITE_RD(RS1)
+  NEXT_INST
+.CKB_VM_ASM_LABEL_OP_CLMUL:
+  DECODE_R
+  ldr RS1, REGISTER_ADDRESS(RS1)
+  ldr RS2, REGISTER_ADDRESS(RS2)
+  mov TEMP4, 0
+  mov TEMP3, 0
+.clmul_branch:
+  mov TEMP1, RS1
+  mov TEMP2, RS2
+  lsl TEMP1, TEMP1, TEMP4
+  lsr TEMP2, TEMP2, TEMP4
+  eor TEMP1, TEMP1, TEMP3
+  and TEMP2, TEMP2, 1
+  csel TEMP3, TEMP1, TEMP3, ne
```

Comment:
It's necessary to you to add `cmp TEMP2, 0` before `csel`.

Same bugs in `clmulh` and `clmulr`

Test case

```
.global _start
_start:
    li a0, 10
    li a1, 20
    clmul a0, a0, a1
    add a0, a0, 120
    li a7, 93
    ecall
```